std/traits.d

```diff
@@ -5364,16 +5369,23 @@ enum bool isNumeric(T) = is(NumericTypeOf!T) && !isAggregateType!T;
 /**
 Detect whether $(D T) is a scalar type (a built-in numeric, character or boolean type).
  */
-enum bool isScalarType(T) = isNumeric!T || isSomeChar!T || isBoolean!T;
+enum bool isScalarType(T) = is(T : real);
```

Comment:
This is wrong. This tests whether the type is implicitly convertible to `real`, and through `alias this`, it's quite possible for a struct or class to be implicitly convertible to `real`.